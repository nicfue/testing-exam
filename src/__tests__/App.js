import React from 'react';
import { shallow, mount, render } from 'enzyme';
import App from '../components/App';
import fakePosts from '..//fakePosts';

it('should render <App /> without user', () => {
  const wrapper = shallow(<App />);
  const user = wrapper.state().currentPersona;
  expect(user).toBe('');
});

it('should render <App /> with user', () => {
  const wrapper = shallow(<App />);
  const persona = 'Nisse';
  wrapper.setState({'currentPersona': persona});
  const user = wrapper.state().currentPersona;
  expect(user).toBe('Nisse');
});

it('Should render App default page', () => {
  const wrapper = shallow(<App />);
  const { currentPage } = wrapper.state();
  expect(currentPage).toBe('home');
});