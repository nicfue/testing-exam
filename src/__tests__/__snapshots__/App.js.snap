// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tesing <App /> button snapshot should match danger default button 1`] = `
<div
  className="mt-8 mx-auto sm:w-full md:w-3/4 lg:w-2/3 "
>
  <Button
    className="absolute pin-l pin-t mt-6 ml-2"
    danger={false}
    onClick={[Function]}
  >
    Talk to a real human
  </Button>
  <PersonaSwitcher
    changePersona={[Function]}
    currentPersona="Nisse"
  />
  <Posts
    currentPersona="Nisse"
  />
</div>
`;
